From patchwork Fri Oct  2 14:31:35 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Christian Hewitt <christianshewitt@gmail.com>
X-Patchwork-Id: 11813649
Return-Path: 
 <SRS0=6DlO=DJ=lists.infradead.org=linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E779B618
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Fri,  2 Oct 2020 14:32:28 +0000 (UTC)
Received: from merlin.infradead.org (merlin.infradead.org [205.233.59.134])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id B4CCD206B2
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Fri,  2 Oct 2020 14:32:28 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=lists.infradead.org
 header.i=@lists.infradead.org header.b="DMEIlmrt";
	dkim=fail reason="signature verification failed" (2048-bit key)
 header.d=gmail.com header.i=@gmail.com header.b="AaUyGK84"
DMARC-Filter: OpenDMARC Filter v1.3.2 mail.kernel.org B4CCD206B2
Authentication-Results: mail.kernel.org;
 dmarc=fail (p=none dis=none) header.from=gmail.com
Authentication-Results: mail.kernel.org;
 spf=none
 smtp.mailfrom=linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=merlin.20170209; h=Sender:Content-Transfer-Encoding:
	Content-Type:MIME-Version:Cc:List-Subscribe:List-Help:List-Post:List-Archive:
	List-Unsubscribe:List-Id:References:In-Reply-To:Message-Id:Date:Subject:To:
	From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:List-Owner;
	bh=x1RPJNOsamtG/NJ+Z9xutLN6+UJRt5h/0oft6PI/TnY=; b=DMEIlmrtmCvf2brdvoRkzgh6UR
	I5MfI4nEzuPiRhLMovnsEIpJtgW1+CO0/IOsdqsjumA8DMUMxTSIFMEcCHEq2annVfqO20YhQqCWQ
	3P5s8/BSpnP0cdiAoSBDvu/NJVNAtjxwZgIfcnBahSKwViVIyarUgsSx5vUk2pviGYJhMxbISBWNj
	h0cdqMBWT5w36XqRJ9E+osVrg57d1lWuJKHmYfS4EG2Iu1Sk5OLfCsxkQubCMQzbMbXhItAjwSN14
	Yih+jbReabKwg55zz8Z+urGYRqhjXOR8KcNASkdubTrAL5fDvRaVTU4hCRQylDVRjc14lqLi4xywT
	i8lCf42A==;
Received: from localhost ([::1] helo=merlin.infradead.org)
	by merlin.infradead.org with esmtp (Exim 4.92.3 #3 (Red Hat Linux))
	id 1kOM6a-0001X5-Lr; Fri, 02 Oct 2020 14:32:16 +0000
Received: from mail-wm1-x335.google.com ([2a00:1450:4864:20::335])
 by merlin.infradead.org with esmtps (Exim 4.92.3 #3 (Red Hat Linux))
 id 1kOM6G-0001Lp-KC; Fri, 02 Oct 2020 14:31:57 +0000
Received: by mail-wm1-x335.google.com with SMTP id w2so1913185wmi.1;
 Fri, 02 Oct 2020 07:31:56 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references;
 bh=6vYF1uN6y1MEmfx/aSCmf3xT19pRJOVeWursMQGhdC8=;
 b=AaUyGK84xBFyPVOg0YZiTXYH5egnyUTDIGSp4B3YsaOkc9U3nnFYHcwiwY3ia2ZJd0
 zJjULreOxlvZEtIwUPLlPum3+szJpSsWj7kv7xETTJqISiBgjiQXMa5GSJKusmOmRibo
 Wsr7VJiQkaqQCPXqCr2FYuRfpYTH7pdIyrnjbPoWt4sPobl7CVYSfAXOqf3RGF5U6f1Q
 OCybM9VfznmPRqUuCcQe2vry+0WGUk7pj9HCOj6gU0H8l48UUzCOJs/2+adCV6S2whs6
 QcZPwp2MaJyAAURazMnjYt9KOtfw/mYobKtWrdixyRn/BGMKxUzCigAb+HmPkW4R7jmg
 xUXQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references;
 bh=6vYF1uN6y1MEmfx/aSCmf3xT19pRJOVeWursMQGhdC8=;
 b=ot08dI7IlOwk1CQwmT3PWWPtl7bswnMm5g2P42Jo2HQvk8vIx4RadPhGzvNHhYs4O7
 AxlS3rjFUv0H15w/ISTlshlX78Zh/+mszSyyeKStOOuCycHuldnB6SsHTKP8uxrSSKCh
 XYThBQd4Bl0wZlwwlTsZVOBV4jKr88dLpvUGcehBsqd2XwlXaVBrSQIXxxTHxJhu9ZG+
 KVBTiuBG/S2vT/7h8XZDtbPqx3UNUwsc4JMHnMJSjvCv1dfLJ5yc+1pE0lkywxVYa1lc
 btHyv1UHn9V0Nv3AGsp69jUGQdWQMuJ44qFEZYoXWWEZ+K+/vKEY0wJN44UqxKG+PZyj
 0fAg==
X-Gm-Message-State: AOAM530IjJn4MqiVo/iOpnPWY9MubNZvhTnL8PcbjP2CD/jP86j4CBUL
 UnSzBAI3QkSxZks+4IiaWFo=
X-Google-Smtp-Source: 
 ABdhPJzq2KLlz7Op/+6RpflhE0nKkBEmL5fs38a+YROdfaB2AoGUhFbZYrhx5MjHepSHaJ41Q7AYdA==
X-Received: by 2002:a1c:9e0e:: with SMTP id h14mr3202779wme.18.1601649115216;
 Fri, 02 Oct 2020 07:31:55 -0700 (PDT)
Received: from localhost.localdomain ([87.200.95.144])
 by smtp.gmail.com with ESMTPSA id w7sm1907392wrm.92.2020.10.02.07.31.53
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Fri, 02 Oct 2020 07:31:54 -0700 (PDT)
From: Christian Hewitt <christianshewitt@gmail.com>
To: Rob Herring <robh+dt@kernel.org>, Mark Rutland <mark.rutland@arm.com>,
 Kevin Hilman <khilman@baylibre.com>, devicetree@vger.kernel.org,
 linux-arm-kernel@lists.infradead.org, linux-amlogic@lists.infradead.org,
 linux-kernel@vger.kernel.org
Subject: [PATCH v2 04/10] arm64: dts: meson: add audio playback to odroid-c2
Date: Fri,  2 Oct 2020 14:31:35 +0000
Message-Id: <20201002143141.14870-5-christianshewitt@gmail.com>
X-Mailer: git-send-email 2.17.1
In-Reply-To: <20201002143141.14870-1-christianshewitt@gmail.com>
References: <20201002143141.14870-1-christianshewitt@gmail.com>
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20201002_103156_680438_EFA440A1 
X-CRM114-Status: GOOD (  13.84  )
X-Spam-Score: -0.2 (/)
X-Spam-Report: SpamAssassin version 3.4.4 on merlin.infradead.org summary:
 Content analysis details:   (-0.2 points)
 pts rule name              description
 ---- ----------------------
 --------------------------------------------------
 -0.0 RCVD_IN_DNSWL_NONE     RBL: Sender listed at https://www.dnswl.org/,
 no trust [2a00:1450:4864:20:0:0:0:335 listed in]
 [list.dnswl.org]
 0.0 FREEMAIL_FROM          Sender email is commonly abused enduser mail
 provider [christianshewitt[at]gmail.com]
 -0.0 SPF_PASS               SPF: sender matches SPF record
 0.0 SPF_HELO_NONE          SPF: HELO does not publish an SPF Record
 0.1 DKIM_SIGNED            Message has a DKIM or DK signature,
 not necessarily
 valid
 -0.1 DKIM_VALID_EF          Message has a valid DKIM or DK signature from
 envelope-from domain
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from
 author's domain
 -0.1 DKIM_VALID Message has at least one valid DKIM or DK signature
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: Christian Hewitt <christianshewitt@gmail.com>
MIME-Version: 1.0
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org

Add initial audio support limited to HDMI i2s.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../boot/dts/amlogic/meson-gxbb-odroidc2.dts  | 40 +++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
index 70fcfb7b0683..9d8c919c6e19 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
@@ -9,6 +9,7 @@
 
 #include "meson-gxbb.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/sound/meson-aiu.h>
 
 / {
 	compatible = "hardkernel,odroid-c2", "amlogic,meson-gxbb";
@@ -172,6 +173,45 @@
 			};
 		};
 	};
+
+	sound {
+		compatible = "amlogic,gx-sound-card";
+		model = "GXBB-ODROID-C2";
+		assigned-clocks = <&clkc CLKID_MPLL0>,
+				  <&clkc CLKID_MPLL1>,
+				  <&clkc CLKID_MPLL2>;
+		assigned-clock-parents = <0>, <0>, <0>;
+		assigned-clock-rates = <294912000>,
+				       <270950400>,
+				       <393216000>;
+		status = "okay";
+
+		dai-link-0 {
+			sound-dai = <&aiu AIU_CPU CPU_I2S_FIFO>;
+		};
+
+		dai-link-1 {
+			sound-dai = <&aiu AIU_CPU CPU_I2S_ENCODER>;
+			dai-format = "i2s";
+			mclk-fs = <256>;
+
+			codec-0 {
+				sound-dai = <&aiu AIU_HDMI CTRL_I2S>;
+			};
+		};
+
+		dai-link-2 {
+			sound-dai = <&aiu AIU_HDMI CTRL_OUT>;
+
+			codec-0 {
+				sound-dai = <&hdmi_tx>;
+			};
+		};
+	};
+};
+
+&aiu {
+	status = "okay";
 };
 
 &cec_AO {
